src/main/resources/Development-documentation/gerrit-sceanrios.md

```diff
@@ -0,0 +1,224 @@
+# Gerrit scenarios
+
+These scenarios describes possible use cases that needs consideration during the
+implementation and update of the plugin
+
+<!-- TOC -->
+
+- [Gerrit scenarios](#gerrit-scenarios)
+  - [Create patch set - CASE1](#create-patch-set---case1)
+  - [Create patch set with rebase - CASE2](#create-patch-set-with-rebase---case2)
+  - [Create patch set with merge - CASE3](#create-patch-set-with-merge---case3)
+  - [Create patch set no previous events - CASE4](#create-patch-set-no-previous-events---case4)
+
+<!-- /TOC -->
+
+## Create patch set - CASE1
+
+Preconditions:
+
+- Previous review submits to  the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]`
+- Receives comments from reviewer
+- Updates code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]`
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+
+Events sent:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
```

Comment:
I think we discussed not to have 'PREVIOUS_VERSION' link.

Reply:
Hmm, for me it feels more logical to have the E2 event link only with PREVIOUS_VERSION to E1 and not the BASE link to E0. 
Since the BASE link exists from E1 event pointing to E0 and if E2 links to E1 with PREVIOUS_VERSION then there is no need for BASE link to E0 from E2 right?
Then each event only contains one type of link and you would follow each event backwards in the chain ... I don't know which is preferred though...

Reply:
But as we discussed. Do we really need to link to previous patchsets that doesn't really provide any valuable information? Isn't it better to just use base links for each new patchset and nothing more?

Reply:
This is a discussion that should continue. Although the scenarios may still have PREVIOUS_VESRIONS for SCC.

Reply:
The current thinking concludes that we should have both PREVIOUS_VERSION and BASE. Please also check bc32435